### PR TITLE
Add re-testing solutions documentation

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: building/tracks/README.md [hunk 1]
```diff
@@ -92,5 +92,32 @@ csharp
...
-└── [config.json](/docs/building/tracks)
+└── config.json
 </pre>
+
+## Re-testing solutions
+
+When you merge a track PR that touches an exercise, it triggers the latest published iteration of students' solutions to be re-tested.
+For popular exercises, this is a _very_ expensive operation (70,000 test runs for Python Hello World as an extreme!).
+
+We've added some further rules today to try and avoid this behaviour. I'll explain the logic in both directions:
```
Comment (iHiD) [inline building/tracks/README.md]:
```suggestion
We encourage you to try and doing this unnecessarily.
```

File: building/tracks/README.md [hunk 2]
```diff
@@ -92,5 +92,32 @@ csharp
...
+Solutions **will not** be retested if:
+
+- You add `[no important files changed]` into the PR's commit body.
+- You only touch docs, hints, `.meta`, or other files that users don't interact with.
+
+Solutions **will** be re-tested if:
+
+1. The PR was merged **without** adding the text `[no important files changed]` into the PR's commit body.
+2. The PR touches one of the following files for an exercise (as specified in its `.meta/config.json` file):
+
```
Comment (iHiD) [inline building/tracks/README.md:113]:
```suggestion
```

File: building/tracks/README.md [hunk 3]
```diff
@@ -92,5 +92,32 @@ csharp
...
+- test files
+- editor files
+- invalidator files
```
Comment (iHiD) [inline building/tracks/README.md]:
```suggestion
  - test files
  - editor files
  - invalidator files
```

File: building/tracks/README.md [hunk 4]
```diff
@@ -92,5 +92,32 @@ csharp
...
+
+When you merge a track PR that touches an exercise, it triggers the latest published iteration of students' solutions to be re-tested.
+For popular exercises, this is a _very_ expensive operation (70,000 test runs for Python Hello World as an extreme!).
+
+We've added some further rules today to try and avoid this behaviour. I'll explain the logic in both directions:
+
+Solutions **will not** be retested if:
+
+- You add `[no important files changed]` into the PR's commit body.
+- You only touch docs, hints, `.meta`, or other files that users don't interact with.
```
Comment (ee7) [inline building/tracks/README.md]:
```suggestion
Solutions **will not** be retested if the merged commit either:

- only touches docs, hints, `.meta`, or other files that users don't interact with
- or contains `[no important files changed]` in the commit body.
```

File: building/tracks/README.md [hunk 5]
```diff
@@ -92,5 +92,32 @@ csharp
...
+
+Solutions **will** be re-tested if:
+
+1. The PR was merged **without** adding the text `[no important files changed]` into the PR's commit body.
+2. The PR touches one of the following files for an exercise (as specified in its `.meta/config.json` file):
```
Comment (ee7) [inline building/tracks/README.md]:
```suggestion
Solutions **will** be re-tested if the merged commit both:

- lacks `[no important files changed]` in the commit body
- and touches one of the following files for an exercise (as specified in its `.meta/config.json` file):
```

**Review by ee7 — COMMENTED** [review]
Great to have this feature.

I've tried to suggest clarifications to the wording of the conditions. There was some ambiguity whether they were AND or OR.

Can we improve the flag name? The goal is to opt out of retesting every user solution when an important file is changed in an unimportant way, right? "No important files changed" is just wrong, because the point is to change important files :)

For example, in https://github.com/exercism/python/pull/3373 @BethanyG wants to change the comment at the top of every Python test file. It's confusing for that commit message to contain "no important files changed".

How about one (or any) of these?

1. `[do not retest every user solution]`
1. `[don't retest every user solution]`
1. `[exercism: do not retest every user solution]`
1. `[exercism: don't retest every user solution]`

Then the effect of the flag is self evident, rather than just something that could be mistaken as a message for a reviewer.

I like option 3. Especially if there's the tiniest potential for us to ever add another flag.

**Pull request merged**